godot-core/src/builtin/meta/mod.rs

```diff
@@ -82,4 +97,147 @@ impl PropertyInfo {
         }
     }
+
+    pub fn empty_sys() -> sys::GDExtensionPropertyInfo {
+        use crate::obj::EngineEnum as _;
+
+        sys::GDExtensionPropertyInfo {
+            type_: VariantType::Nil.sys(),
+            name: std::ptr::null_mut(),
+            class_name: std::ptr::null_mut(),
+            hint: global::PropertyHint::PROPERTY_HINT_NONE.ord() as u32,
+            hint_string: std::ptr::null_mut(),
+            usage: global::PropertyUsageFlags::PROPERTY_USAGE_NONE.ord() as u32,
+        }
+    }
+}
+
+// ----------------------------------------------------------------------------------------------------------------------------------------------
+/// Rusty abstraction of sys::GDExtensionClassMethodInfo
+
+/// Info relating to an argument or return type in a method.
+pub struct MethodArgumentInfo {
+    info: PropertyInfo,
+    metadata: sys::GDExtensionClassMethodArgumentMetadata,
+}
+
+/// All info needed to
+pub struct MethodInfo {
+    class_name: ClassName,
+    method_name: StringName,
+    call_func: sys::GDExtensionClassMethodCall,
+    ptrcall_func: sys::GDExtensionClassMethodPtrCall,
+    method_flags: MethodFlags,
+    return_value: Option<MethodArgumentInfo>,
+    arguments: Vec<MethodArgumentInfo>,
+    default_arguments: Vec<Variant>,
+}
+
+impl MethodInfo {
+    pub fn from_signature<S: VarcallSignatureTuple>(
+        class_name: ClassName,
+        method_name: StringName,
+        call_func: sys::GDExtensionClassMethodCall,
+        ptrcall_func: sys::GDExtensionClassMethodPtrCall,
+        method_flags: MethodFlags,
+        param_names: &[&str],
+        default_arguments: Vec<Variant>,
+    ) -> Self {
+        let return_value = S::return_info();
+        let mut arguments = Vec::new();
+
+        assert_eq!(
+            param_names.len(),
+            S::ARGUMENTS_LEN,
+            "`param_names` should contain one name for each parameter"
+        );
+
+        for (i, name) in param_names.into_iter().enumerate().take(S::ARGUMENTS_LEN) {
+            arguments.push(S::argument_info(i, name).unwrap())
+        }
+
+        assert!(
+            default_arguments.len() <= arguments.len(),
+            "cannot have more default arguments than arguments"
+        );
+
+        Self {
+            class_name,
+            method_name,
+            call_func,
+            ptrcall_func,
+            method_flags,
+            return_value,
+            arguments,
+            default_arguments,
+        }
+    }
+
+    pub fn register_extension_class_method(&self) {
+        use crate::obj::EngineEnum as _;
+
+        let (has_return_value, return_value_info, return_value_metadata) = match &self.return_value
+        {
+            Some(info) => (true, Some(info.info.clone()), info.metadata),
+            None => (false, None, 0),
+        };
+
+        let mut return_value_sys = return_value_info
+            .as_ref()
+            .map(|info| info.property_sys())
+            .unwrap_or(PropertyInfo::empty_sys());
```

Comment:
`return_value_info` doesn't need to be cloned, it could just be referenced, no? Then you don't need the `Clone` derive, either.

Also, `has_return_value` can simply be used as `return_value_info.is_some()` when initializing `sys::GDExtensionClassMethodInfo`.